torch/csrc/jit/runtime/interpreter.cpp

```diff
@@ -1495,21 +1516,36 @@ struct InterpreterStateImpl : c10::intrusive_ptr_target {
             ++af.pc;
           } break;
           case WARN: {
+            // Keeps track of which WARN instruction has been executed before,
+            // we only want to execute each WARN once to match default Python
+            // warning behavior.
+            // TODO TODO this set should be graph specific, rather than global.
+            bool need_warn = true;
+            if (inst.X != -1) {
+              auto inserted = warned_indices_.insert(inst.X);
```

Comment:
you probably need a lock for it because afaiu the same `Code` can run concurrently from several threads

Reply:
Added.